bin/user/updatecheck.py

```diff
@@ -0,0 +1,203 @@
+                    else:
+                        raise Exception("Version not found in remote skin.conf")
+
+            except (urllib.error.URLError, urllib.error.HTTPError, Exception) as e:
+                if attempt < RETRIES:
+                    log.debug("Retry in %d seconds... (%s)", DELAY, str(e))
+                    time.sleep(DELAY)
+                else:
+                    log.info("Failed to fetch latest version from GitHub after %d attempts", RETRIES)
+                    log.debug("Last error: %s", e)
+                    return None
+
+        return None
+
+    def _compare_versions(self, current_ver, latest_ver):
+        """Compare versions and return update information based on check mode"""
+        try:
+            if HAVE_PACKAGING:
+                current = version.parse(current_ver)
+                latest = version.parse(latest_ver)
+            else:
+                current = LooseVersion(current_ver)
+                latest = LooseVersion(latest_ver)
+
+            if latest <= current:
+                log.debug("Current version %s is up to date (latest: %s)", current_ver, latest_ver)
+                return None
+
+            log.debug("Current version %s latest version: %s)", current_ver, latest_ver)
+
+            # Determine if we should show this update based on the check mode
+            if self.update_check_mode == "minor":
+                # Only show if major or minor version changed
+                if HAVE_PACKAGING:
+                    if current.major != latest.major or current.minor != latest.minor:
+                        return self._create_update_info(current_ver, latest_ver, "minor")
+                else:
+                    # Fallback: parse version manually for major.minor comparison
+                    current_parts = current_ver.split('.')
+                    latest_parts = latest_ver.split('.')
+                    if (len(current_parts) >= 2 and len(latest_parts) >= 2 and
+                        (current_parts[0] != latest_parts[0] or current_parts[1] != latest_parts[1])):
+                        return self._create_update_info(current_ver, latest_ver, "minor")
+            elif self.update_check_mode == "patch":
+                # Show all updates including patches
+                return self._create_update_info(current_ver, latest_ver, "patch")
+
+            return None
+
+        except Exception as e:
+            log.debug("Error comparing versions %s vs %s: %s", current_ver, latest_ver, e)
+            return None
+
+    def _create_update_info(self, current_ver, latest_ver, update_type):
+        """Create update information dictionary"""
+        return {
+            "available": True,
+            "current_version": current_ver,
+            "latest_version": latest_ver,
+            "update_type": update_type,
+            "github_url": "https://github.com/seehase/neowx-material",
+            "release_url": "https://github.com/seehase/neowx-material/releases"
```

Comment:
May be little confusing if check is done against master, but link sends user to releases. It can happen that it will show that there is new version, but after following link, there would be no release yet. Github offers API to check latest release, maybe it would be better to do check agains it ?

https://api.github.com/repos/seehase/neowx-material/releases/latest

It returns json and one attribute is tag_name, what about checking against it ? But not sure what about bug fix version as they usually are not released as separated release.